openff/toolkit/topology/molecule.py

```diff
@@ -2151,6 +2151,12 @@ def to_networkx(data: FrozenMolecule | nx.Graph) -> nx.Graph:
                 return data.to_networkx()
 
             elif isinstance(data, nx.Graph):
+                for node in data:
+                    h_counter = -1
+                    for neighbor in data.neighbors(node):
+                        if data.nodes[neighbor]['atomic_number'] == 1:
+                            data.nodes[neighbor]['atomic_number'] = h_counter
+                            h_counter -= 1
```

Comment:
Does this depend on the order of the `data.neighbors(node)` iterator (in problematic ways)? I will try to think of an antagonistic test case tomorrow, if possible